### PR TITLE
Check if PublishedPropertyType modelType is null

### LINKED ISSUE #13515: Model's builder doesn't nicely catch a type error making it hard to debug an data type issue.
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.3.2

### Bug summary

Model's builder throws an exception and fails to build the models.

The way we discovered this was by creating a Data List type using Contentment, there may be other ways to replicate the same error. When trying to generate Models using Models Builder, we receive the following error:

```
Object reference not set to an instance of an object.
at Umbraco.Cms.Core.Models.PublishedContent.ModelType.MapToName(Type type, Dictionary`2 map, Boolean dictionaryIsInvariant)
```

Which as you can see, this isn't particularly useful to help debug where the issue might be, it would be great if this error could be caught and a more helpful error message thrown so the developer knows which datatype is causing the issue.

### Specifics

The current error is thrown on this line:

https://github.com/umbraco/Umbraco-CMS/blob/e7fae144c2bdedfd78ccd53c27ec97b3fa41604b/src/Umbraco.Core/Models/PublishedContent/ModelType.cs#L229

### Steps to reproduce

To replicate this error, create a Contentment Data List data type and use a .NET Enumeration, don't select a library or class to enumerate, then save it. Add this to a document type and then try to build the models.

### Expected result / actual result

We would expect when building models that any error in data types were handled gracefully and hopefully bubbled up to the UI, or worst case scenario they should be thrown with useful debugging information so a developer can easily fix the issue without having to spend an hour trying to debug :)

It would be great for this error to indicate the name of the data type that caused the issue.

### PR DESCRIPTION
**Pull request by CyberReiter**
If there's an existing issue for this PR then this fixes #13515 

### Description
Adding a null check for the modeltype right after the PropertyValueConverter has been called to check if a type has been set.
That the type returned is null should never happen when using default PropertyValueConverters but if you are using a custom PropertyValueConverter there could be a chance that null is returned.
Sadly we only have the Id of the datatype available, but thats still better than nothing.

### CODE REVIEW

File: src/Umbraco.Core/Models/PublishedContent/PublishedPropertyType.cs [hunk 1]
```diff
@@ -190,7 +190,13 @@ private void InitializeLocked()
...
 
             _cacheLevel = _converter?.GetPropertyCacheLevel(this) ?? PropertyCacheLevel.Snapshot;
-            _modelClrType = _converter == null ? typeof (object) : _converter.GetPropertyValueType(this);
+            Type modelClrType = _converter == null ? typeof (object) : _converter.GetPropertyValueType(this);
+            if(modelClrType == null)
+            {
+                throw new ArgumentNullException($"Unable to map DataType with Id {DataType.Id} to a C# type.");
+            }
+
+            _modelClrType= modelClrType;
```
Comment (JasonElkin) [inline src/Umbraco.Core/Models/PublishedContent/PublishedPropertyType.cs:199]: We don't really want or need to throw here. 

There's already a convention that if the converter itself is null we fall back to returning `typeof (object)`.

`GetPropertyValueType` shouldn't really be returning null here anyway (this suggests a bug with the Property Value Converter itself), but if it does, it makes sense to fall back to the same convention and return `typeof (object)`.

```suggestion
            _modelClrType = _converter?.GetPropertyValueType(this) ?? typeof(object);
```

**Review by JasonElkin — CHANGES_REQUESTED** [review]
Thanks so much @CyberReiter for taking a look at this.

Though this is a bug/problem with the Property Value Converter, we don't really want to throw here because we can gracefully fall back to `typeof(object)`. 

Strictly speaking, there's no need for a Property Value Converter for a given property, hence the fallback, so it makes sense to treat a null type the same way we would when there is no converter at all.

If you're happy to change it to return `typeof(object)` (I've added an example) that would be amazing, otherwise let me know and I can do it and we'll get this fix merged in ASAP.

File: src/Umbraco.Core/Models/PublishedContent/PublishedPropertyType.cs (hunk 1 above)
Comment (CyberReiter) [reply src/Umbraco.Core/Models/PublishedContent/PublishedPropertyType.cs]: Hey @JasonElkin 
Thank you very much for the great input here aswell as on the issue itself!
It absolutely makes sense to use the same convention here, so I changed it as you suggested.

Comment (JasonElkin) [reply src/Umbraco.Core/Models/PublishedContent/PublishedPropertyType.cs]: Super! Thanks for that @CyberReiter I'll re-test and get back to you later.

**Review by JasonElkin — APPROVED** [review]
This works like a charm. Thanks @CyberReiter.

**Comment by nul800sebastiaan** [comment]
Cherry picked for 10.5 in 5fcbd4e4cd883f5e48189a4d116ea387990acb4d

**Pull request merged**